packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -535,6 +536,11 @@ class SemanticsData with Diagnosticable {
   /// The reading direction is given by [textDirection].
   final String tooltip;
 
+  /// The hierarchy level of the widget as a header.
+  ///
+  /// This should only be set when [SemanticsFlag.isHeader] is set.
```

Comment:
is this true? the isHeader may be closer to banner in aria.

Reply:
Yes, a widget with isHeader attribute gets a "role=heading" aria attribute, not a "banner" role. For that reason I consider this change is needed, a "heading" role needs an "aria-level". Webs without this do not pass accessibility static analysis checks and we really need this for a specific project.

Reply:
> isHeader attribute gets a "role=heading

is this already the case in web? I remember when I last revisited this, only the Appbar has the isHeader = true. and Appbar is closer to banner in html https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/banner_role

The heading, ie h1, h2, h3 seams to be a different thing in html

cc @yjbanov

Reply:
This doesn't seem to be answered yet

Reply:
Hi @chunhtai, sorry for replying so late. Yes, I don't know what's the purpose of having that parameter enabled for the AppBar, but I can confirm applying the "header" attribute has the effect I mention, I tested it many times, also you can see evidence in this code (part of the engine)

![image](https://github.com/flutter/flutter/assets/3387118/5958d4cc-6df3-4d0b-9db1-760d972a9ffe)

Reply:
I think the code you link is incorrect. @yjbanov should it use banner in engine instead?

Reply:
Yes, I think flutter web's usage of `role=heading` is incorrect. It was my mistake, as I was misunderstood what `isHeader` meant. "banner" seems closer to what it means in Flutter.

Reply:
If that's the case and we change isHeader to "banner", there won't be a way to add headings and levels in flutter web. Having a heading structure is an important feature in order to have decent web accessibility. Should we have another attribute for this?

Reply:
`isHeader` is named after the similar widget in mobile platforms.

https://developer.apple.com/documentation/uikit/uiaccessibilitytraitheader?language=objc

Android called it heading

https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo#isHeading() 

It seems to me the mobile `header` or `heading` is the equivalent `banner` in web.

> Should we have another attribute for this?

Isn't the heading level enough for this? we need to make sure we document that the heading level has nothing to do with the `isHeader` property.

Reply:
This seems not addressed yet?

Reply:
> This seems not addressed yet?

Sorry, not yet, but I will address it. If I understood correctly the idea is to separate both attributes, changing "heading" property to be used to establish a "banner" role in web, and dedicate the headingLevel attribute to add the element as a header role with its corresponding level.